### PR TITLE
feat: Implement electron.DesktopCapturer.SetSkipCursor

### PR DESCRIPTION
**Pull request by CapOM**
#### Checklist

- [x] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] relevant documentation is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [ ] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes: Can disable or re-enable cursor capturer while the stream is running

### CODE REVIEW

**Comment by CapOM** [comment]
> These are fairly chonky patches, can you attempt to upstream them so we don't have to take that burden?

It would not be possible to upstream it in the current state. I think the only way would be to start implementing `MediaTrackSupportedConstraints.cursor` (values `always`, `never`, `motion`) , see the w3c spec https://www.w3.org/TR/screen-capture/#cursorcaptureconstraint . But that would be to support `getDisplayMedia` which is not currently supported by electron: https://github.com/electron/electron/issues/16513#issuecomment-640854503
So that would reduce this PR but not sure how much.

There are some technical details here but essentially skipping the cursor on a particular source would affect all calls to getUserMedia or getDisplayMedia on that particular source (i.e. a particular app window A)

This is because the compositing of the cursor and the frame is done in the webrtc capturer. Some interfaces would need to be changed in chromium/media first to allow passing a frame and the cursor as a meta overlay. So that consumers of the capturer could decide or not to composite the frame with the cursor independently of each others for the same source.

That said currently this PR allows to show and hide the cursor on the fly. And skipping the cursor on a source app window A does not affect capturing the cursor on another source B. It only affects if there are multiple captures of the same source app window A at the same time.
As mentioned above, it would be very difficult to avoid that corner case when implementing `MediaTrackSupportedConstraints.cursor`. Also the use case were we would want to skip and show the cursor on app window A , should be rare.

Note that even if this PR touches lot of files but only a few lines per files and the changes are quite redundant.

**Comment by pavlobu** [comment]
Great work! 
highly waiting for it to appear in new version

File: docs/api/desktop-capturer.md [hunk 1]
```diff
@@ -97,6 +97,13 @@ which can detected by [`systemPreferences.getMediaAccessStatus`].
 [`navigator.mediaDevices.getUserMedia`]: https://developer.mozilla.org/en/docs/Web/API/MediaDevices/getUserMedia
 [`systemPreferences.getMediaAccessStatus`]: system-preferences.md#systempreferencesgetmediaaccessstatusmediatype-macos
 
+### `desktopCapturer.SetSkipCursor(sourceId, skip)`
```
Comment (miniak) [inline docs/api/desktop-capturer.md:100]:
```suggestion
### `desktopCapturer.setSkipCursor(sourceId, skip)`
```

**Comment by apodolny** [comment]
Would love to see this merged — @nornagon @miniak let me know if there's anything I can do to help.

**Comment by nornagon** [comment]
Looks like this PR somehow got messed up. I can't review it as is.

**Comment by apodolny** [comment]
@miniak do you have time to resolve the issues here?

**Comment by ns-jisorce** [comment]
Hi, I will try to rebase my PR against master later this year

**Comment by apodolny** [comment]
@CapOM, are you able to rebase against master? Let me know if there's anything I can do to help.

**Comment by adityak368** [comment]
Any update on this? Let me know if I can help

**Comment by CapOM** [comment]
Yes please go ahead with the rebase if you have some time, I do not have the bandwidth for now. Thanks

**Comment by tristaaan** [comment]
I tried rebasing on master and I had a rather difficult time with it and ended up cherry-picking the commit. 
https://github.com/tristaaan/electron/tree/set_skip_cursor_rebased

**Comment by apodolny** [comment]
Seems like something has happened to this PR again :(

**Comment by nornagon** [comment]
I'm closing this PR as it's unreviewable in its current state, please open a new PR if you're interested in continuing development!

**Comment by apodolny** [comment]
> I tried rebasing on master and I had a rather difficult time with it and ended up cherry-picking the commit.
> https://github.com/tristaaan/electron/tree/set_skip_cursor_rebased

Do you want to make a new PR based on your change?

**Comment by adityak368** [comment]
@CapOM Created a new PR instead of rebase, which seemed to be the easier way. I think the webrtc patch needs to be updated as it failed to apply for me.

**Pull request closed without merging**